### ISSUE TITLE
multi-row bookmarks: Can't drag and drop things into sub-folders

### ISSUE THREAD

**Issue by Mr-Personality** [MrOtherGuy/firefox-csshacks#37]
When scrolling is enabled (when the number of rows is limited), you can't drag and drop anything into sub-folders.

You can drag things into top-level folders, but when you try to drag into a sub-folder (moving the cursor down), the whole bookmarks bar scrolls down and the place you're trying to drag to moves with it.

Therefore I cannot currently use the scrolling/limited number of columns feature.

**Comment by MrOtherGuy** [comment]
I've researched this a bit and I think this cannot be fixed. So if you really need to reorder bookmarks very often then your best option is probably to use bookmarks sidebar.

I'm going to leave this open in case some solution comes up, but I wouldn't be holding my breath.

**Comment by Mr-Personality** [comment]
Thanks for the reply.

Yes, I do reorder my bookmarks all the time. I am a very organised person. In fact, the way I create all my bookmarks is to drag the address bar directly into the correct folder. I find that the quickest way.

This is not a problem though, I've realized that I can simply change the script to increase the number of rows to 5 (the parameter "multirow-bmb-n-rows"). For years I used a UserScript (which recently broke) allowing me to have 5 rows, and that's what works best for me.

So my problem is solved, thank you. And thank you for the excellent script! :-)

As for using the bookmarks sidebar, I have five personal reasons not to use that:

1. First and foremost, I use a vertical tab bar (Vertical Tabs Reloaded) which leaves no space for the bookmarks bar, and would probably conflict with it.

2. The bookmarks bar shows all my bookmarks at once, but the bookmarks toolbar requires me to scroll across 3 pages to see all my bookmarks.

3. The bookmarks bar shows all my bookmarks arranged across 5 rows, allowing me to easily remember the general location of everything and thus accessing it quickly. But with the bookmarks toolbar, nothing has a 'location', it's just a long list.

4. I also much prefer the horizontal arrangement of the bookmarks bar. Tall lists are fine for some things, such as lengthy lists of items, but when it comes to organising and arranging, I find a horizontal configuration much more natural and pleasing, since human vision is wide not tall.

5. I also like all the bookmarks being at the top of the screen, because it's the "starting point", and things logically flow from top to bottom.
